### ISSUE TITLE
NeDB Memory issues and possible solution.

### ISSUE THREAD

**Issue by danyg** [louischatriot/nedb#382]
Hi,
As part of a project of mine, I'm experiencing with some memory issues in NeDB.

So, the problem is I'm working with datastores which are growing to reach Gigabytes and, I have a few of them.
First I would like to ask if there is any way to UNLOAD the memory datastores or "deprecate them" to be moved by the OS to SWAP or something like that.

In the other hand, I'm creating an implementation which will decorate a Datastore object which in fact will be running in a forked process.
After a while with any I/O that process will be killed, freeing the memory. If for some reason all of them are needed in memory, and there is no more Physical Memory available, in this way the OS should, and probably will, move a few of them to the SWAP.

We are always talking of No Critical uses, and this Datastores are only storing information which is read and write by a Web / Web Service, for a bunch users, so the delays produced in the load of the Datastores on demand, is completely acceptable.

So here is the Issue I've created to keep tracking of it... if somebody want to know more about it... or wants to contribute to the cause :P

[cucumber-statistics #7](https://github.com/danyg/cucumber-statistics/issues/7)

Thanks!

**Comment by lukasbestle** [comment]
I think you should use MongoDB or another standalone database. Embedded databases aren't meant to be used with GBs of data.

**Comment by louischatriot** [comment]
Yup, definitely sounds like a usecase for a large scale solution such as
Mongo !
Le 22 févr. 2016 12:54 PM, "Lukas Bestle" <notifications@github.com> a
écrit :

**Comment by darrrk** [comment]
Maybe you scale your database on several computers, each of which interact with db over mounted sshfs partition.

**Comment by darrrk** [comment]
Or try to plug more RAM and start nodejs process with flag `--max-old-space-size=n`, where n - RAM in Mb.
